cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -144,6 +148,24 @@ private void addPlugin(PrintWriter writer, Plugin plugin)    {
         }
     }
 
+    private void validateUrlPresence(DownloadableArtifact art) {
+        if (art.getUrl() == null) {
+            throw new InvalidConfigurationException(art.getType() + " artifact is missing an URL.");
+        }
+    }
+
+    private void validateGavPresence(MavenArtifact art) {
```

Comment:
Wouldn't it be simpler to have a `validate` method on `Artifact` which was implemented by `DownloadableArtifact` to check the URL and by `MavenArtifact` to check the GAV? Or is `InvalidConfigurationException` not in `api`?

Reply:
`ICE` is not in the `api`